usvg/src/image.rs

```diff
@@ -37,6 +39,22 @@ impl std::fmt::Debug for ImageKind {
     }
 }
 
+/// Functions that accept various representations of `xlink:href` value of the `<image`>
+/// element and return ImageKind that holds reference to the image buffer determined by these functions.
+#[allow(clippy::type_complexity)]
+pub struct ImageHrefResolver {
+    /// Resolver function that will be used if `xlink:href` is a DataUrl with encoded base64 string.
+    pub resolve_data: Box<dyn Fn(&str, Arc<Vec<u8>>, &OptionsRef) -> Option<ImageKind>>,
+
+    /// Resolver function that will be used to handle arbitrary string in `xlink:href`.
+    pub resolve_string: Box<dyn Fn(&str, &OptionsRef) -> Option<ImageKind>>,
+}
+
+impl std::fmt::Debug for ImageHrefResolver {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("ImageHrefResolver closures (..)")
```

Comment:
`ImageHrefResolver { .. }` is more [correct](https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html#method.finish_non_exhaustive) (this method is available since 1.53, we cannot use it yet)